### ISSUE TITLE
[Feature] Disable adjustments to scrolling

### ISSUE THREAD

**Issue by ciriousjoker** [linearmouse/linearmouse#346]
### OS

macOS 13.2

### LinearMouse

v0.7.7

### Is your feature request related to a problem?

I'm using Logitech Options+, which has its own controls for scroll speed and smooth scrolling, which I can set on a per-app basis as well as globally.

LinearMouse enforces whatever I set in it globally and seems to override Logitech Options+ in this regard.

### Describe the solution you'd like

I'd like to have a toggle in LinearMouse that disables any adjustments to how scrolling works. I only use LinearMouse to disable pointer acceleration.

### Describe alternatives you've considered

Uninstall LinearMouse, but that's not really an option since it's the only way to reliably disable pointer acceleration.

**Comment by lujjjh** [comment]
> LinearMouse enforces whatever I set in it globally and seems to override Logitech Options+ in this regard.

If you have changed the scrolling acceleration/speed in LinearMouse, you can click "Revert to system defaults" to revert the settings to the default values, which means that LinearMouse will not affect the scrolling behavior.

These settings won't override Logitech Options+'s because scrolling acceleration and speed are not properties stored in system settings or device drivers. What's under the hood is a modification chain. For example, LinearMouse intercepts scrolling events, modifies the distance, and passes the modified events to the next application that intercepts scrolling events (say Logitech Options+).

**Comment by ciriousjoker** [comment]
Got it. I guess I'll figure something out... Thanks for the explanation!

**Comment by ciriousjoker** [comment]
@lujjjh 

**Perhaps related to:**
- #107 

LinearMouse seems to break Logitech Options+' scroll speed setting. All other settings work (invert, smooth scrolling, smart shift etc.). As soon as I kill LinearMouse, scroll speed also works again.

While LinearMouse is running, changing the scroll speed in Logitech Options+ does nothing.

**Comment by lujjjh** [comment]
I'm not sure how Logitech Options+ changes the scrolling speed. It doesn't make sense that LinearMouse would the scrolling speed setting but leave other scrolling settings working.

> While LinearMouse is running, changing the scroll speed in Logitech Options+ does nothing.

Neither the global one nor the application-specific one works?

Is this reproducible with previous releases of LinearMouse?

**Comment by ciriousjoker** [comment]
@lujjjh 

global/app specific makes no difference.

Switched to v0.2.1, options+ working again
0.5.3: options+ working again
0.6.2: options+ working again

in all of these cases, the scroll settings in LinearMouse make no difference, not even invert scrolling.
it seems like install order makes a difference?

went back to v0.7.7:
- inverted scrolling started working again
- options+ is broken again
apparently install order doesnt make a difference here

v0.7.3:
- doesnt launch on the first try (second try works): <img width="324" alt="Bildschirm­foto 2023-02-14 um 10 55 39" src="https://user-images.githubusercontent.com/16866223/218701285-cfbe7c76-3b86-4d27-8bde-8bc96df3f61b.png">
- invert scrolling works
- option+ speed works
- as soon as I turn on "linear scrolling", options+ breaks again (disabling it fixes this)

---

I got no time to try out more versions right now. Let me know if I should test something specific.

**Comment by lujjjh** [comment]
Thanks for testing so many versions.

It sounds like Options+ can detect the scrolling distance change made by other applications even if the change appears to be a no-op (e.g. multiply the original scrolling distance by 1). Although I still don't know how Options+ does this, it's the only explanation I can think of.

Please try [this build](https://github.com/linearmouse/linearmouse/suites/10968699068/artifacts/555507541) (built from #348). I hope it solves this issue.

**Comment by ciriousjoker** [comment]
@lujjjh 

Wait with merging for now. I'd like to test if the last production version is different from this test build. Hopefully I get to it in the next two hours

**Comment by ciriousjoker** [comment]
@lujjjh 
- Seems like v0.7.7 is no different from the test build. 
- "accelerated" works with options+, but pixels/lines breaks it

Is it even possible to use Options+' speed setting with LinearMouse's lines/pixels? The PR didn't make a difference here.

**Comment by lujjjh** [comment]
> "accelerated" works with options+, but pixels/lines breaks it

This is the expected behavior.

So I didn't fully understand what you meant. Scrolling by pixels/lines is designed to scroll a certain distance each time the wheel gets turned, so it's possible that the Option+' speed setting being overwritten.

Can you describe your use cases? If you prefer pixel/line scrolling and want to select different scrolling distances for different applications, you can refer to [app matching](https://github.com/linearmouse/linearmouse/blob/main/Documentation/Configuration.md#app-matching).

**Comment by ciriousjoker** [comment]
@lujjjh 

I much prefer the line scrolling because it interpolates better. But it's too fast if I also turn on smooth scrolling

**Comment by lujjjh** [comment]
I have no idea what the smooth scrolling of Options+ looks like. Does it work like [Mos](https://github.com/Caldis/Mos/blob/master/README.enUS.md) and [Smooze](https://smooze.co)?

Enabling smooth scrolling in other apps while enabling linear scrolling in LinearMouse may be problematic: if the scroll event reaches LinearMouse later, the distance of the scrolling interpolated by other apps may be overwritten. If the order is reversed, it may work fine. But we can't ensure the order of events, so I would suggest to set the scroll distance and smooth scrolling settings in the same app. If Options+ does not support it, consider using Mos or Smooze.

In short, try to avoid configuring similar features in multiple apps at the same time. I'll try to keep LinearMouse as compatible as possible with other similar apps, but it's not possible to completely avoid conflicts.

**Comment by ciriousjoker** [comment]
@lujjjh 

Will try that later. Options+ smooth scrolling seems to fire smaller events in quick succession, but LinearMouse converts that to one line per Mini-Event, which is way too fast.

**Comment by ciriousjoker** [comment]
I didn't try the suggested solutions with Smooze and Mos, but I'm closing this anyway since I figured out a set of settings in Options+ and LinearMouse that are satisfying enough to use. Maybe I'll try the other two things at some point, but for now I'm happy that it works at all to a reasonable degree.

Thanks for your help and for this great tool!
